### PR TITLE
Remove transaction type relation from Listing

### PR DESCRIPTION
**Pull request by rap1ds**

### CODE REVIEW

File: app/controllers/listings_controller.rb [hunk 1]
```diff
@@ -156,6 +156,7 @@ def new
       shape = get_shape(Maybe(params)[:transaction_type].to_i.or_else(nil))
 
       # PaymentRegistrationGuard needs this to be set before posting
+      @listing.transaction_type_id = shape[:transaction_type_id]
```
Comment (rap1ds) [inline app/controllers/listings_controller.rb]: Remove this, it's added two times

File: spec/mailers/person_mailer_spec.rb [hunk 2]
```diff
@@ -104,7 +104,9 @@
     @test_person.update_attributes({ "given_name" => "Teppo", "family_name" => "Testaaja" })
 
     transition = FactoryGirl.build(:transaction_transition, to_state: "confirmed")
-    listing = FactoryGirl.build(:listing, author: @test_person)
+    listing = FactoryGirl.build(:listing,
+                                transaction_type_id: 123,
```
Comment (rap1ds) [inline spec/mailers/person_mailer_spec.rb]: Change this

File: spec/services/transaction_service/paypal_events_spec.rb [hunk 3]
```diff
@@ -50,7 +50,9 @@ def create_test_transaction(opts)
   before(:each) do
     @cid = 3
     @payer = FactoryGirl.create(:payer)
-    @listing = FactoryGirl.create(:listing, price: Money.new(45000, "EUR"))
+    @listing = FactoryGirl.create(:listing,
+                                  price: Money.new(45000, "EUR"),
+                                  transaction_type_id: 123)
```
Comment (rap1ds) [inline spec/services/transaction_service/paypal_events_spec.rb]: Change this

File: features/step_definitions/listing_steps.rb [hunk 4]
```diff
@@ -2,9 +2,24 @@
...
+    tt_model = find_transaction_type_by_name(transaction_type) if transaction_type
+    shape = ListingService::API::Api.shapes.get(community_id: @current_community.id, transaction_type_id: tt_model.id)[:data]
+
+    opts[:transaction_process_id] = shape[:transaction_process_id]
+    opts[:transaction_type_id] = shape[:transaction_type_id]
+    opts[:shape_name_tr_key] = shape[:name_tr_key]
+    opts[:action_button_tr_key] = shape[:action_button_tr_key]
+  else
+    opts[:transaction_process_id] = 9999
+    opts[:transaction_type_id] = 9999
```
Comment (rap1ds) [inline features/step_definitions/listing_steps.rb]: Change this

File: features/step_definitions/admin_transaction_steps.rb [hunk 5]
```diff
@@ -35,10 +35,16 @@ def build_transaction(transaction_data)
     }
     sum = transaction_data[:sum].to_i * 100 unless transaction_data[:sum].empty?
 
+
+    listing = FactoryGirl.build(:listing,
+                                title: transaction_data[:listing],
+                                author: author,
+                                transaction_type_id: 9999)
```
Comment (rap1ds) [inline features/step_definitions/admin_transaction_steps.rb]: Change this

File: app/indices/listing_index.rb [hunk 6]
```diff
@@ -22,7 +22,7 @@
   has created_at, updated_at
   has sort_date
   has category(:id), :as => :category_id
-  has transaction_type(:id), :as => :transaction_type_id
```
Comment (sktoiva) [inline app/indices/listing_index.rb]: Check if this needs reconfiguration of flying-sphinx.

File: app/models/listing.rb [hunk 7]
```diff
@@ -333,7 +327,7 @@ def closed?
   def as_json(options = {})
     # This is currently optimized for the needs of the map, so if extending, make a separate JSON mode, and keep map data at minimum
     hash = {
-      :listing_type => ListingShapeHelper.transaction_type_to_direction(self.transaction_type), # deprecated
+      :listing_type => ListingShapeHelper.transaction_type_to_direction(transaction_type_id), # deprecated
```
Comment (sktoiva) [inline app/models/listing.rb]: Should be `transaction_type_id_to_direction`

File: app/models/transaction_type.rb [hunk 8]
```diff
@@ -63,7 +62,7 @@ def display_name(locale)
       .get(community_id, {
         translation_keys: [name_tr_key],
         locales: community.locales
-      })
+           })
```
Comment (sktoiva) [inline app/models/transaction_type.rb]: Fix indent.

File: features/step_definitions/admin_category_steps.rb [hunk 9]
```diff
@@ -226,8 +218,8 @@ def find_transaction_type_by_name(transaction_type_name)
 end
 
 When /^I toggle transaction type "(.*?)"$/ do |transaction_type_name|
-  transaction_type = find_transaction_type_by_name(transaction_type_name)
-  find(:css, "#transaction_type_checkbox_#{transaction_type.id}").click
+  shape = find_shape(name: transaction_type_name)
```
Comment (sktoiva) [inline features/step_definitions/admin_category_steps.rb]: Remember to fix finding by name. Now it breaks with two clashing names. 

So, don't allow same shape names for same locale inside a community.

**Pull request merged**